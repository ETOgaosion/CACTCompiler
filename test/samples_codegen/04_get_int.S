iobuff:
	.zero 80
.text
.align 1
.globl	main
.type	main, @function
main:
	sd	ra, -8(sp)
	mv	t0, zero
	sw	t0, -12(sp)
	addi	sp, sp, -16
	call	get_int
	addi	sp, sp, 16
	sw	a0, -16(sp)
	lw	t0, -16(sp)
	sw	t0, -12(sp)
	lw	a0, -12(sp)
	addi	sp, sp, -16
	call	print_int
	addi	sp, sp, 16
	mv	a0, zero
	ld	ra, -8(sp)
	ret
