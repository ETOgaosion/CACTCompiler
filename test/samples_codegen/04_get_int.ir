============ global var ================
============ imm value ================
============ functions ================
----------- funcDef -----------
main() => 2:
----------- paramVars -----------
----------- localVars -----------
a := symbol type: 1; data type: 2; initValue: 0

----------- tempVars ------------
t_0:= symbol type: TEMP VAR; data type: 2
----------- Codes ------------
main:
	a = 0;
	call get_int, frame size: 16; 
	t_0 =  return ;
	t_1 <= a;
	t_1 = t_0;
	addparam a;
	call print_int, frame size: 16; 
	return 0;

