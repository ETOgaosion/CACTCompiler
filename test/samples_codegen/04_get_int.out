bbl loader
please enter an int number:
